feat(kb)/add circumference task 1

Aliaksei9 opened the pull request:
Добавил задачу тема Окружность и круг

Commit:
feat(kb)/add circumference task 1

pYro-sudo commented on kb/task_formalization/Circumference1.gwf:
Когда доделаешь, нужно будет сложить в папку, как было сделано у меня в пул реквесте. 

То есть, берешь свою тему, переводишь на английский, а потом с таким названием создаешь папку. Потом, файл нужно будет назвать так: перевод на английский и номер задания, но номер не из книги, а каким ты его по счету кладешь в папку, то есть, если первым, то 1, если 5 по счету, то 5. Что-то должно быть такое:
```<topic_in_english>_task_<number_of_task>.gwf```.

А теперь к самому графу. 

- Во-первых, надо было найти ломаную OKMN, то есть, что есть такая ломаная, которая принадлежит этой окружности, а еще этой ломаной принадлежат отрезки OK, ON, MK, MN, но про них, что они не посчитаны, оставь, думаю.
- Во-вторых, у тебя названия окружности и центра окружности совпадают, возьми букву T, например.
- В-третьих, понятие длины не материя, кружочек штриховой нужен, а не с птичкой.
- Угол подмножество ломаных, так что ```concept_angle``` тоже надо штрихом.
- ```concept_number``` замени на ```number``` везде.
- Ломаная будет не как ты перевел, что получилась сломанная линия, а ```polygonal chain```
![image](https://github.com/user-attachments/assets/4ac07017-ff03-4913-9f65-49ae424feb23)
- Тут ты промазал, убедись, чтобы оно было к узлу, а не стрелке (это где второй шаг).
![image](https://github.com/user-attachments/assets/e870e0a5-2a1a-4fe8-85ba-4319b59cb016)
- Тоже промахнулся.
![image](https://github.com/user-attachments/assets/3be5a64e-ec61-4942-a13b-74da7feb95ec)
- Это отношения, так что стрелка должная быть соответствующая. Посмотри везде, чтобы правильно было.
- Не забудь везде в каждой ```sc-link```e посмотреть, чтобы точки стояли, и нужное количество, где у тебя не числа, а твои умные мысли, а еще писать то, что какая-то штука равна xxx см нужно так: ```xxх (см|cm)```, то есть, например, ```12 cm```, и про единицы измерения тоже не забудь.
- Диаметр чего, радиус чего указывать, то есть, в твоем случае окружности, там в книге даже написано везде, потому что можно найти радиус графа, например, диаметр разбиения.
![image](https://github.com/user-attachments/assets/b2cda192-7215-4d18-8307-ed0aa55b8ce5)
- Проверь, чтобы каждая из таких узлов отношений была ими и помечена.
![image](https://github.com/user-attachments/assets/594a3f4a-2969-420d-92c0-819ba5ea7323)
- На этом этапе ты нашел длины хорд тоже. Либо разбей этапы, либо укажи про них здесь тоже. И у окружности есть имя, как и радиуса.
![image](https://github.com/user-attachments/assets/fbc1e7f6-369f-42ba-89dc-f19783b9d8df)
- Вот этого тоже быть не должно, у тебя центр окружности не действие.
- Ну, может, я уже погнал прям далеко, но на последней фотке это не формула диаметра, а про отношение диаметра и радиуса.
![image](https://github.com/user-attachments/assets/663b19d7-3905-40c9-a738-0aab7ab23bde)
- Тут как аргумент не может передаться структура, которая будет на этом этапе сгенерирована, то есть, этой ломаной тут быть не должно, ты вообще окружность туда должен передать, как и в прошлом действии, а про ломаную в условии формализовать.

Если какие-то трудности в переводе, то залезь в Википедию, там выбери английский язык, и всё. В файле я ничего не правил. Может, я что-то упустил, но на то тут и Наташа тоже.

pYro-sudo replied:
А, ещё, наверное, оттого, что ломаная замкнутая, надо будет сказать, что она принадлежит классу замкнутых ломаных, а те уже ломаным

Malina09 replied:
по поводу второго комментария, дизлайк так как не нужно прописывать классы в классы, это всё уже прописано в самой БЗ, дублировать не нужно

pYro-sudo reviewed (commented):
читай, короче

Commit:
review fixes

review fixes 2

review fixes 3

fixes task 1

fixes task 1

Circumderence in circle

T

-the

polygonal chain in bz

polygonal chain in bz

fix formulas and concept_geometric_point

pYro-sudo reviewed (commented):
Только еще одно но. Там точки есть у радиусов, которые ОК и ОN, так вот, потому что они, по идее, из неба свалились, нужно сказать, что им принадлежат точки O, K & O, N соответственно. Только вот вопрос, что, по идее, в условии напрямую не сказано, что OKMN принадлежат все те отрезки, но это подразумевается. Так что в решении, наверное, вот нужно сказать, что вот эти отрезки принадлежат ломаной. Но и условие тогда нужно переделать, сказав, что есть эти четыре точки, и они принадлежат OKMN(это про условие).
![image](https://github.com/user-attachments/assets/48d6df0e-3109-43f0-903d-833a7ecde042)
и тут надо будет убрать, что ты лишний раз сказал, что есть радиус(на первом этапе решения).

И надо, думаю, сказать, что те две хорды равны радиусу, то написать, что у хорды есть длина, и у радиуса radius тоже, и сделать так, чтобы у них длины были одинаковые(это в условии)

Malina09 reviewed (commented):
- А N и K не angular_point? 
![image](https://github.com/user-attachments/assets/c1ab1502-cd87-4c48-8863-67f1ed861827)
- так в учебнике написано? просто звучит довольно странно, может быть лучше что-то вроде "Радиус равен половине диаметра" или "Диаметр равен двум радиусам"
![image](https://github.com/user-attachments/assets/881cf951-4cd2-46c2-9c76-c24368ab62e9)
- 17.5 уже было прописано в первом пункте, не нужно везде прописывать идентификатор, уберите во втором пункте
![image](https://github.com/user-attachments/assets/e3b96897-6d83-4373-a556-7f6005d11e36)
- также вот эта вся часть уже была прописана в условии, больше не нужно этого делать, убрать из второго действия это и подобные вещи
![image](https://github.com/user-attachments/assets/52115403-4f29-4308-ac6b-a735d509a864)

pYro-sudo commented:
Там мы убирали с Лешей, Наташ, больше этого быть не должно, если все правильно закоммичено было

Aliaksei9 commented:
Я исправил определение

Malina09 reviewed (commented):
- в БЗ вроде как есть уже замкнутая ломанная, и называется по-другому, надо посмотреть, как это вообще правильно называть, и поменять либо тут, либо в самой БЗ везде (сделать карточку на это)
![image](https://github.com/user-attachments/assets/dad94af3-dc72-4c8a-aed7-c541bce7c72c)
![image](https://github.com/user-attachments/assets/e0a01605-1ecf-4a69-86be-75f6897761e0)
- concept_circumference имело в виду окружность ведь, а не длина окружности? если да, то должно быть concept_circle и, соответственно, не nrel_center_of_the_circumference, а nrel_center_of_circle 
- должно быть rrel_1, а не nrel_1, в двух действиях сейчас nrel
![image](https://github.com/user-attachments/assets/bfbb4218-43d2-4d13-ae73-bda6baf2bfa9)
- вот это уже в условии было расписано 
![image](https://github.com/user-attachments/assets/57b839a8-fa18-4842-afed-8e9ff83da42c)
- дважды один и тот же узел привязан, но вообще это не нужно, так как досаточно один раз где-то указать, что 17.5 это number 
![image](https://github.com/user-attachments/assets/a1c116b3-3435-48ad-817a-fe46182e46ca)
- везде перепроверь повторяющиеся конструкции, вероятно ON надо указать, что является радиусом, а вот сам радиус уже не надо, это есть в условии 
![image](https://github.com/user-attachments/assets/26fd624f-12cc-4e76-82ad-57718520978e)

Malina09 reviewed (commented):
аналогично тому, что писала выше, вот тут уже не нужно уточнять, что это окружность, в условии прописывали, в действиях нужно убрать это
![image](https://github.com/user-attachments/assets/173451a5-51fb-482d-b312-8ed7e019f2ad)
остальное вроде теперь норм

Aliaksei9 commented:
Finally?

Malina09 commented:
Я бы хотела, чтобы может быть ещё Ваня глянул

pYro-sudo commented:
кстати, как тут быть, то не знаю, но это не совсем правильное замечание про ломаные. Полилиния не совсем про это, потому что это конструкция, которая допускает и кривые. И тут имелась ломаная, которая не может допускать кривых, а она правильно определена, как на вики. По поводу формулы радиуса, то она может и применяться для вычисления диаметра, так что было бы про отношение точнее. Ну, про последнее согласен.

pYro-sudo reviewed (commented):
ну, да, остается, чтобы ты сделал вот те замечания, котрые я оспорить не смог, как по мне. И возможно, но я не уверен, что это так обязательно, но нужно будет немножко упростить конструкции в условии. Просто тогда нужно разбить на графы поменьше, но чтобы не было тавтологий в формализации. То есть, условно, что какие-то точки принадлежат какой-то прямой, не нужно писать повторно.

pYro-sudo reviewed (approved):
все ок, если всех устраивает оформление

Malina09 commented:
closed_polyline на polygonal_chain будете в этом PR менять или в другом?

pYro-sudo commented:
может, потом?

Malina09 commented on kb/plane_geometry/absolute/concept_closed_polygonal_chain.scs:
```diff
@@ -0,0 +1,21 @@
+		(*
+		-> rrel_example: 
+		"file://images/concept_closed_polygonal_chain.png"(*=> nrel_format: format_png;;*);;
+
```
```suggestion

```

Malina09 reviewed (commented):
![image](https://github.com/user-attachments/assets/b086fd06-e64f-42ec-a899-d5798b2513f1)
у меня ещё вот столько файлов осталось с closed_polyline

Malina09 commented:
![image](https://github.com/user-attachments/assets/d9749c30-5477-4ff9-8738-9970e301eb19)
а это что за интересная конструкция? в шаблоне не так сделано

Malina09 committed:
fix(kb): change name for  closed polygonal chain

Malina09 commented:
исправьте мой комментарий, и наведите порядок в коммитах, остальное вроде ок

aChainsmoker commented:
Двойку уже пару раз формализовывали, ещё раз не надо
![image](https://github.com/user-attachments/assets/55093215-04f8-45f2-bd36-d7e0947cd27b)

В базе этот класс называется concept_geometric_point
![image](https://github.com/user-attachments/assets/b903629f-cf0c-4e7b-99e1-4519c6f1f085)


также исправь объяснения, про что тебе уже писали раннее, и можно мержить, я думаю

Malina09 reviewed (approved):
вроде бы выглядит хорошо

pYro-sudo reviewed (commented):
![image](https://github.com/user-attachments/assets/5cbe9b40-0e45-4834-b488-cc6cf7be2a9b)
формулирока не ```красткая```, а ```краткая```

pYro-sudo committed:
Update s_sd_convex_polygons_ru.html

pYro-sudo reviewed (approved):
так и быть, соизволю исправить одну букву

Aliaksei9 commented:
Прошу прощения что пришлось. Были сильные проблемы со здоровьем.
